lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -324,6 +330,61 @@ class TwilioConnectionTransport extends StateMachine {
     }
     return false;
   }
+
+  /**
+   * @returns {void}
+   */
+  _setSession(session, sessionTimeout) {
+    this._session = session;
+    this._sessionTimeoutMS = sessionTimeout * 1000;
+  }
+
+  /**
+   * Determines if we should attempt reconnect.
+   * returns a Promise to wait on before attempting to
+   * reconnect. returns null if its not okay to reconnect.
+   * @private
+   * @returns {Promise<void>}
+   */
+  _getReconnectTimer() {
+    if (this._sessionTimeoutMS === 0) {
+      // this means either we have never connected.
+      // or we timed out while trying to reconnect
+      // In either case we do not want to reconnect.
+      return null;
+    }
+
+    // start session timer
+    if (!this._sessionTimer) {
+      this._sessionTimer = setTimeout(() => {
```

Comment:
Please use the Timeout class in lib/util.

Reply:
I looked into it, but the implementation of Timeout calls `.clear()` before calling callback function. That does not work in this case because we need to check here if the timer has been cleared while we were waiting. 

https://github.com/twilio/twilio-video.js/blob/master/lib/util/timeout.js#L67

we could workaround it, by creating a new timer object everytime, but I think it's much easier w/o the `Timeout` object

Reply:
@makarandp0 Instead of checking whether the timeout object exists, you can use a private boolean property:
```js
this._sessionTimedOut = false;
this._sessionTimeout = new Timeout(() => this._sessionTimedOut = true, SESSION_TIMEOUT_MS);
```

Reply:
@manjeshbhargav - I agree we could make `Timeout` work by introducing additional variable, but I am not sure that is better solution. I feel basic `setTimeout`  does job effectively here.